automation/android/espresso/build.gradle

```diff
@@ -0,0 +1,197 @@
+    }
+}
+
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+
+// Borrowed from http://stackoverflow.com/questions/17097263/automatically-versioning-android-project-from-git-describe-with-android-studio-g
+def getVersionCode = {
+    try {
+        def bitrise_build_number = System.getenv("BITRISE_BUILD_NUMBER")
+        if (bitrise_build_number != null)
+        {
+            // If we're on bitrise, then just use the BITRISE_BUILD_NUMBER
+            return Integer.parseInt(bitrise_build_number)
+        }
+        else {
+            // Otherwise, use the git build number
+            def stdout = new ByteArrayOutputStream()
+            exec {
+                commandLine 'git', 'rev-list', '--first-parent', '--count', 'master'
+                standardOutput = stdout
+            }
+            return Integer.parseInt(stdout.toString().trim())
+        }
+    }
+    catch (ignored) {
+        return -1
+    }
+}
+
+def getProperty(key, defaultValue) {
+    rootProject.ext.hasProperty(key) ? rootProject.ext[key] : defaultValue
+}
+
+def _compileSdkVersion = getProperty('compileSdkVersion', 26)
+def _buildToolsVersion = getProperty('buildToolsVersion', '28.0.3')
+def _minSdkVersion = getProperty('minSdkVersion', '17')
+def _targetSdkVersion = getProperty('targetSdkVersion', 26)
+def _versionCode = getVersionCode()
+def _versionName = "1.0." + _versionCode
+
+android {
+    compileSdkVersion _compileSdkVersion
+    buildToolsVersion _buildToolsVersion
+
+    defaultConfig {
+        minSdkVersion _minSdkVersion
+        targetSdkVersion _targetSdkVersion
+        versionCode _versionCode
+        versionName _versionName
+
+        def dittoMode = findProperty("dittoMode") ?: "play"
+        buildConfigField "String", "GLOBAL_DITTO_MODE", "\"$dittoMode\""
+    }
+
+    buildTypes {
+        debug {
+            minifyEnabled false
+            shrinkResources false
+        }
+        release {
+            minifyEnabled false
+            shrinkResources false
+        }
+    }
+
+    lintOptions {
+        abortOnError false
+    }
+
+    packagingOptions {
+        // Required for Espresso.
+        exclude 'META-INF/maven/com.google.guava/guava/pom.properties'
+        exclude 'META-INF/maven/com.google.guava/guava/pom.xml'
+    }
+
+    // Android Studio doesn't pick up kotlin folder automatically
+    // https://kotlinlang.org/docs/reference/using-gradle.html
+    sourceSets {
+        main.java.srcDirs += 'src/main/kotlin'
+    }
+}
+
+dependencies {
+    // last update: Sept 30 2017
+    // old versions: $ANDROID_HOME/extras/android/m2repository/com/android/support/test/
+    // new versions: https://maven.google.com
+    //
+    // match versions from:
+    // https://developer.android.com/topic/libraries/testing-support-library/release-notes.html
+    //
+    // https://maven.google.com/com/android/support/test/espresso/espresso-core/3.0.0/espresso-core-3.0.0.pom
+
+    def runnerVersion = "1.1.0-beta01"
```

Comment:
these versions look out of date, there should be new versions for the espresso dependencies

Reply:
This is what we had defined in our android-uno Versions object.  I'll look around to see if there are better/newer versions.  (Note that the Android Studio is not complaining that there versions are out of date, as it seems to normally do.)

Reply:
> (Note that the Android Studio is not complaining that there versions are out of date, as it seems to normally do.)

right, it won't do that for any dynamically defined versions.

Reply:
find the latest stable versions here:
https://developer.android.com/training/testing/release-notes